README.md

```diff
@@ -22,9 +39,275 @@ Or install it yourself as:
 
     $ gem install accessly
 
+Add the ActiveRecord Migrations:
+
+    $ rails g accessly:install
+
 ## Usage
 
-TODO: Write usage instructions here
+You can use the Accessly gem directly to grant | revoke | check permissions.  We recommend the use of 'Policies' which will be covered in this README.
+Checkout our [API docs](http://www.rubydoc.info/gems/accessly) for more info on using the API directly
+
+We use Accessly with policies in mind to capture everything we want to know about a specific permission set. Let's take a look at some examples:
+
+### Basic Action Policy
+
+```ruby
+class ApplicationFeaturePolicy < Accessly::Policy::Base
+
+  actions(
+    view_dashboard: 1,
+    view_super_secret_page: 2,
+    view_double_secret_probation_page: 3
+  )
+
+end
+```
+
+With this policy we can `grant` permissions to a user
+
+```ruby
+ApplicationFeaturePolicy.new(user).grant!(:view_super_secret_page)
+```
+
+In our `SuperSecretPageController`, we can check whether the user has permission to view that page with
+
+```ruby
+ApplicationFeaturePolicy.new(user).view_super_secret_page?
+# or
+ApplicationFeaturePolicy.new(user).can?(:view_super_secret_page)
+```
+
+At any point in time we can revoke permissions with
+
+```ruby
+ApplicationFeaturePolicy.new(user).revoke(:view_super_secret_page)
+```
+
+### Basic Action on Object Policy
+
+We can grant permissions to `actors` on other `objects` in our application with a policy like:
+
+```ruby
+class UserPolicy < Accessly::Policy::Base
+
+  actions_on_objects(
+    view: 1,
+    edit: 2,
+    destroy: 3
+  )
+
+  def self.namespace
+    User.name
+  end
+
+  def self.model_scope
+    User.all
+  end
+end
+```
+
+We differentiate permissions by a `namespace` which by default is the name of your policy class.  However,
+It may be necessary to override the default behavior represented in the above example.
+
+Accessly can return a relation of ids on an object for a given actor's permission grants.  `Accessly::Policy::Base` requires
+that you implement `self.model_scope` with an `ActiveRecord` scope so the `list` api can return an `ActiveRecord::Relation`
+
+With this policy we can `grant` permissions for a user to do an action on another user object.
+
+```ruby
+UserPolicy.new(user).grant!(:edit, other_user)
+```
+
+In our `EditUserConroller`, we can check permissions
```

Comment:
I think this is a typo, did you mean `Controller`?

Reply:
I did! 👍